server/node.go

```diff
@@ -292,6 +298,12 @@ func (n *Node) initStores(engines []engine.Engine, stopper *util.Stopper) error
 		n.initNodeID(0)
 	}
 
+	// Now that we know our ClusterID and NodeID, store it in our LocalSender,
+	// which embeds a context.Context as a base context for requests.
+	log.Add(n.lSender,
```

Comment:
It's quite confusing to use the local sender as the argument here. If you really want the context embedded, at least using `n.lSender.Context` here as an argument.